data_retriever.rb

```diff
@@ -0,0 +1,110 @@
+require 'nokogiri'
+require 'rubygems'
+require 'open-uri'
+require 'watir'
+require 'pry'
+require 'json'
+require_relative "Account.rb"
+require_relative "Transaction.rb"
+
+
+class Data_Retriever
+  def initialize()
+    @available_accounts = []
+    @transactions_records = []
+    @browser = Watir::Browser.new :chrome
+  end
+
+  def browse(url)
+    @browser.goto(url)
+    @browser.div(:id => "oCAPICOM").wait_while_present(&:present?)
+  end
+
+  def login
+    @browser.a(class: 'ng-binding').wait_until(&:present?).click
+    @browser.a(id: 'demo-link').click
+    @browser.element(:id => "step1").wait_until(&:present?)
+  end
+
+  def parsing
+    @page = Nokogiri.HTML(@browser.html)
+    puts "Parsing is done."
+  end
+
+  def gather_credit_info
+    @page.css("#dashboardAccounts tr[id]").each do |row|
+    account_name = row.css("span[bo-bind='row.iban']").text
+    account_currency = row.css("span[bo-bind='row.acyAvlBal | sgCurrency']").text.to_f 
+    account_available_balance = row.css("span[bo-bind='row.ccy']").text
+    credit_accs = Account.new(account_name , account_available_balance , account_currency , "Credit")

```

Comment:
credit_accs means that in this variable you have more than one account and that adds some misunderstanding. Call it credit_account or credit_acc